src/main/java/com/google/sps/servlets/QueryFactory.java

```diff
@@ -0,0 +1,53 @@
+package com.google.sps.servlets; 
+
+import com.google.cloud.spanner.Statement;
+import java.util.List;
+
+final class QueryFactory {
+  
+  private static final QueryFactory instance = new QueryFactory();
+  private static final String GROUP_BY_TABLE_NAMES = ") group by table_name";
+
+  private QueryFactory() {
+    // Private because static class
+  }
+
+  static QueryFactory getInstance() {
+    return instance;
+  }
+
+  //TODO: try to do binding w/ parameters
+  static Statement buildSchemaQuery(String table) {
+    StringBuilder query = new StringBuilder("SELECT column_name, spanner_type, is_nullable ");
+    query.append("FROM information_schema.columns WHERE table_name = '%s'");
```

Comment:
Using StringBuilder would be useful if we have some if conditions or for loops, but since here we always build the query string, and know what it looks like, consider build it directly. 

String query = "SELECT column_name, spanner_type, is_nullable FROM information_schema.columns WHERE table_name = '%s' "

Reply:
Done